### PR TITLE
chore: Add testing for NodeRequirement Drift

### PR DESCRIPTION
**Pull request by engedaam**
Fixes # 

**Description**
- Added e2e testing for node requirement drift

**How was this change tested?**

**Does this change impact docs?**
- [x] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by njtran — COMMENTED** [review]
I think this is the right direction, but I think we need some more cases to test NodeSelectorRequirements changing here, made a similar comment in your karpenter-core PR.

File: test/suites/drift/suite_test.go [hunk 1]
```diff
@@ -289,7 +292,7 @@ var _ = Describe("Drift", Label("AWS"), func() {
 				},
 			})
 		})
-		DescribeTable("provisioner static drift", func(fieldName string, provisionerOption test.ProvisionerOptions) {
+		DescribeTable("provisioner drift", func(fieldName string, provisionerOption test.ProvisionerOptions) {
```
Comment (jonathan-innis) [inline test/suites/drift/suite_test.go:295]:
```suggestion
		DescribeTable("Provisioner Drift", func(fieldName string, provisionerOption test.ProvisionerOptions) {
```

We've generally been capitalizing our test names

File: test/suites/drift/suite_test.go [hunk 2]
```diff
@@ -279,7 +279,10 @@ var _ = Describe("Drift", Label("AWS"), func() {
 				SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
```
Comment (jonathan-innis) [inline test/suites/drift/suite_test.go:279]: This doesn't have a `go.mod` bump so this test is going to fail if it gets merged, right?

Comment (engedaam) [reply test/suites/drift/suite_test.go:279]: Yeah, just waiting to merge the core PR

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**